File: python/old/Project Euler/Euler_179.py
def factors(n):
    t = 0
    for i in range(1,int(n**(0.5))+1):
        if n % i == 0:
            t += 1
            if i**2 <> n:
                t += 1
    return t

total = 1
now = factors(2)
nxt = factors(3)

for i in range(3,10000000):
    now = nxt
    nxt = factors(i+1)
    if now == nxt:
        total += 1

print 'Total: %s' % total

File: c++/Project Euler/Euler_179.c
#include <iostream>
#include <cmath>

using namespace std;

int factors(int n) {
    int total = 0;
    for (int i = 1; i < (int)sqrt(n) + 1; i++) {
        if (n % i == 0) {
            total++;
            if (i * i != n) {
                total++;
            }
        }
    }
    return total;
}

int main()
{
    int total = 1;
    int now = 2;
    int nxt = 2;

    for (int i = 3; i < 10000000; i++) {
        now = nxt;
        nxt = factors(i+1);
        if (now == nxt) {
            total++;
        }
    }

    cout << "Total: " << total;
}
